typing/ctype.ml

```diff
@@ -1148,8 +1150,8 @@ let rec copy ?partial ?keep_names scope ty =
     t.desc <-
       begin match desc with
       | Tconstr (p, tl, _) ->
-          let abbrevs = proper_abbrevs p tl !abbreviations in
-          begin match find_repr p !abbrevs with
+          (*let abbrevs = proper_abbrevs env p tl !abbreviations in*)
```

Comment:
I'm surprised by the removal of `proper_abbrevs` here. I assume it is just because `copy` does not currently take an environment, which `proper_abbrev` now needs. Morally, I think that `copy` should take an environment anyway, since it creates a type at the current level, which is closely tied to the environment. So it would probably be better to add an environment parameter to `copy`.

Reply:
FTR. it did take an environment until 7f0e819bde632d2491ffc3a99e9814343a4daf6f

Reply:
@lpw25 I need a better analysis of the code, but this doesn't seem to make a difference.
Actually, I wonder whether the original version was correct or no. This change returns the code to what it was in ocaml 2.04.
@trefs Indeed, at some point `copy` used the environment, but your scope mechanism removed this need. Since I'm not sure it's really needed here, it seems simpler for now to get rid of some dubious code.